whats_fresh/whats_fresh_api/tests/views/entry/test_inline_product.py

```diff
@@ -0,0 +1,125 @@
+
+    Things tested:
+        URLs reverse correctly
+        The outputted popup form has the correct form fields
+        POSTing "correct" data will result in the creation of a new
+            object with the specified details
+        POSTing data with all fields missing (hitting "save" without entering
+            data) returns the same field with notations of missing fields
+    """
+
+    def setUp(self):
+        user = User.objects.create_user(
+            'temporary', 'temporary@gmail.com', 'temporary')
+        user.save()
+
+        admin_group = Group(name='Administration Users')
+        admin_group.save()
+        user.groups.add(admin_group)
+
+        response = self.client.login(
+            username='temporary', password='temporary')
+        self.assertEqual(response, True)
+
+    def test_not_logged_in(self):
+        self.client.logout()
+
+        response = self.client.get(
+            reverse('edit-vendor', kwargs={'id': '1'}))
+        self.assertRedirects(response, '/login?next=/entry/vendors/1')
+
+    def test_url_endpoint(self):
+        url = reverse('product_ajax')
+        self.assertEqual(url, '/entry/vendors/new/products/new')
+
+    def test_form_fields(self):
+        """
+        Tests to see if the form contains all of the right fields
+        """
+        response = self.client.get(reverse('product_ajax'))
+
+        fields = {'name': 'input', 'variety': 'input', 'story': 'select',
+                  'alt_name': 'input', 'description': 'textarea',
+                  'origin': 'input', 'season': 'input', 'available': 'select',
+                  'market_price': 'input', 'link': 'input',
+                  'image': 'select'}
+        form = response.context['product_form']
+
+        for field in fields:
+            # for the Edit tests, you should be able to access
+            # form[field].value
+            self.assertIn(fields[field], str(form[field]))
+
+    def test_successful_product_creation(self):
+        """
+        POST a proper "new product" command to the server, and see if the
+        new product appears in the database
+        """
+        # Create objects that we'll be setting as the foreign objects for
+        # our test product
+
+        # It needs a story, and we'll want multiple product_preparations to
+        # allow us to test the multi-product logic.
+
+        # We can't predict what the ID of the new product will be, so we can
+        # delete all of the vendors, and then choose the only vendor left
+        # after creation.
+        Product.objects.all().delete()
+
+        Story.objects.create(id=1)
+        Preparation.objects.create(id=1)
+        Preparation.objects.create(id=2)
+        Image.objects.create(id=1)
+
+        # Data that we'll post to the server to get the new Product created
+        inline_product = {'name': 'Salmon', 'variety': 'Pacific', 'story': 1,
+                          'alt_name': 'Pacific Salmon', 'origin': 'The Pacific',  # noqa
```

Comment:
Instead of putting ``# noqa`` on all of these lines, just break it across more lines, putting `alt_name` on its own line.